### ISSUE TITLE
AWS_Profile driver option

### ISSUE THREAD

**Issue by tkeller-moxe** [awslabs/aws-advanced-jdbc-wrapper#775]
### Describe the feature

Ability to set the AWS SSO profile to use in driver settings for use with IAM/Secrets manager authentication

### Use Case

We need a way other then environment to specify AWS profile. The wrapper currently does not support this. 

### Proposed Solution

Hopefully this is just adding a new parameter and updating https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/wrapper/src/main/java/software/amazon/jdbc/plugin/IamAuthConnectionPlugin.java#L233 to take the optional profile in. 

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### The AWS Advanced JDBC Driver version used

2.3.0

### JDK version used

17

### Operating System and version

Windows/Debian

**Comment by brunos-bq** [comment]
Hi @tkeller-moxe ,

Thanks for reaching out and raising this issue.

We'll take a look at this and keep you updated as we investigate.

Thank you for your patience!

**Comment by brunos-bq** [comment]
Hello @tkeller-moxe,

Would you be able to obtain what you need if you could configure Amazon credentials using a CustomHandler?


```
AwsCredentialsManager.setCustomHandler((hostSpec, props) -> DefaultCredentialsProvider.builder().profileName("custom-profile").build());
```

**Comment by tkeller-moxe** [comment]
For my use case I can not implement any custom code. We are using this as a wrapper to enable third party applications to use IAM. 
EG Dbeaver with many local profiles/rds instances to connect to. 

Deployed applications with an IAM role can have explicit permissions granted. The traditional local SSO pattern has you use profiles all with different credential sets. Which means using the environment variables to set this becomes...hard.

**Comment by krzysztofdrys** [comment]
Hi all,

This feature would be very useful for my use case as well. We have a legacy application which we want to migrate to RDS IAM Auth. Switching from one driver to another without any code change looks like the easiest way.

Do I get this right that this issue will be closed by https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/786 ? And we are only waiting for https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/786 to be released?

**Comment by crystall-bitquill** [comment]
Hi @tkeller-moxe and @krzysztofdrys,

To update you, we've merged in #786 which allows users to specify a profile name for AWS credentials through the [`awsProfile` configuration parameter](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/UsingTheJdbcDriver.md#aws-advanced-jdbc-driver-parameters). Additional documentation on credential provider configuration can be found [here](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/custom-configuration/AwsCredentialsConfiguration.md). These changes have been included in the latest release, version 2.3.3. Could you kindly check it out and let us know if this solution works for you?

Thank you!

**Comment by krzysztofdrys** [comment]
Hi @crystall-bitquill! I have just tested this feature in 2.3.3. It does exactly what we need. Thanks for all your hard work!

**Comment by tkeller-moxe** [comment]
Will do!
This is in our pipeline to get tested on our end :) 
Thank you so much :D

**Comment by twglomski** [comment]
I was able to verify that this works correctly:
- With Account A as my default account, and a database that has permissions only to those in Account B, I specified the profile name for Account B with the `awsProfile` variable and was able to make a successful postgres jdbc connection.
- I did notice that if I changed my `awsProfile` to a profile that did **not** have access to that database, the driver would be able to make the connection immediately after making the swap. I believe this is due to the fact that the RDS IAM authentication gets short term credentials from AWS when authenticating, and when those short term credentials expire access is no longer allowed (which I observed). Overall I don't think this is a concern and it looks good for what we're doing! 👍
